### ISSUE TITLE
Property changes to null and other values every run without any actual changes

### ISSUE THREAD

**Issue by mayaaravotwm** [akamai/terraform-provider-akamai#180]
Property changed without any actual changes every build since v0.5.0 

### Terraform Version
Terraform v0.12.25
Akamai v0.7.2

### Affected Resource(s)
- akamai_property
- akamai_property_activation

### Terraform Configuration Files
```hcl

resource "akamai_property" "mobile" {
  name     = "mobile-${data.terraform_remote_state.vpc.outputs.env}"
  contract = data.akamai_contract.contract.id
  cp_code  = data.akamai_cp_code.code.id
  group    = data.akamai_group.group.id
  contact  = ["CONTACT"]
  product  = "prd_Site_Accel"

  hostnames = {
    (var.env == "prod" ? "ADDRESS" : "${data.terraform_remote_state.vpc.outputs.env}-ADDRESS") = "${data.terraform_remote_state.vpc.outputs.env}-ADDRESS"
  }

  is_secure   = true
  rule_format = "v2018-02-27"
  rules       = "${data.akamai_property_rules.rules.json}"

  depends_on = ["akamai_edge_hostname.edge"]
}

resource "akamai_property_activation" "staging" {
  property = "${akamai_property.mobile.id}"
  network  = "STAGING"
  activate = true
  contact  = ["CONTACT"]
  version  = "${akamai_property.mobile.version}"
}

resource "akamai_property_activation" "production" {
  property = "${akamai_property.mobile.id}"
  network  = "PRODUCTION"
  activate = true
  contact  = ["CONTACT"]
  version  = "${akamai_property.mobile.version}"
}

data "akamai_property_rules" "rules" {
  rules { # Default rule
    behavior {
      name = "origin"
      option {
        key   = "cacheKeyHostname"
        value = "ORIGIN_HOSTNAME"
      }
      option {
        key   = "originType"
        value = "CUSTOMER"
      }
      option {
        key   = "hostname"
        value = "1.2.3.4"
      }
      option {
        key   = "forwardHostHeader"
        value = "REQUEST_HOST_HEADER"
      }
      option {
        key   = "compress"
        value = true
      }
      option {
        key   = "enableTrueClientIp"
        value = false
      }
      option {
        key   = "httpPort"
        value = 80
      }
      option {
        key   = "httpsPort"
        value = 443
      }
      option {
        key   = "verificationMode"
        value = "CUSTOM"
      }
      option {
        key   = "originSni"
        value = false
      }
      option {
        key   = "originCertsToHonor"
        value = "STANDARD_CERTIFICATE_AUTHORITIES"
      }
      option {
        key    = "standardCertificateAuthorities"
        values = ["akamai-permissive", "THIRD_PARTY_AMAZON"]
      }
      option {
        key    = "customValidCnValues"
        values = ["{{Origin Hostname}}", "{{Forward Host Header}}"]
      }
    }
    behavior {
      name = "caching"
      option {
        key   = "behavior"
        value = "CACHE_CONTROL_AND_EXPIRES"
      }
      option {
        key   = "mustRevalidate"
        value = true
      }
      option {
        key   = "defaultTtl"
        value = "0h"
      }
      option {
        key   = "honorPrivateEnabled"
        value = false
      }
      option {
        key   = "honorMustrevalidateEnabled"
        value = false
      }
    }
    behavior {
      name = "allowPost"
      option {
        key   = "enabled"
        value = true
      }
      option {
        key   = "allowWithoutContentLength"
        value = false
      }
    }
    behavior {
      name = "report"
      option {
        key   = "logHost"
        value = false
      }
      option {
        key   = "logReferer"
        value = false
      }
      option {
        key   = "logUserAgent"
        value = true
      }
      option {
        key   = "logAcceptLanguage"
        value = false
      }
      option {
        key   = "logCookies"
        value = "OFF"
      }
      option {
        key   = "logCustomLogField"
        value = false
      }
    }
    behavior {
      name = "sureRoute"
      option {
        key   = "enabled"
        value = false
      }
      option {
        key   = "srDownloadLinkTitle"
        value = ""
      }
    }
    behavior {
      name = "tieredDistribution"
      option {
        key   = "enabled"
        value = true
      }
    }
    behavior {
      name = "prefetch"
      option {
        key   = "enabled"
        value = true
      }
    }
    behavior {
      name = "realUserMonitoring"
      option {
        key   = "enabled"
        value = false
      }
    }
    behavior {
      name = "http2"
      option {
        key   = "enabled"
        value = ""
      }
    }
    behavior {
      name = "allowTransferEncoding"
      option {
        key   = "enabled"
        value = true
      }
    }
    rule {
      name = "Redirect to HTTPS"
      behavior {
        name = "redirect"
        option {
          key   = "queryString"
          value = "APPEND"
        }
        option {
          key   = "responseCode"
          value = 301
        }
        option {
          key   = "destinationHostname"
          value = "SAME_AS_REQUEST"
        }
        option {
          key   = "destinationPath"
          value = "SAME_AS_REQUEST"
        }
        option {
          key   = "destinationProtocol"
          value = "HTTPS"
        }
        option {
          key   = "mobileDefaultChoice"
          value = "DEFAULT"
        }
      }
      criteria {
        name = "requestProtocol"
        option {
          key   = "value"
          value = "HTTP"
        }
      }
    }
    rule {
      name = (var.env == "prod" ? "ADDRESS " : "${data.terraform_remote_state.vpc.outputs.env}-ADDRESS")
      criteria {
        name = "hostname"
        option {
          key   = "matchOperator"
          value = "IS_ONE_OF"
        }
        option {
          key    = "values"
          values = [(var.env == "prod" ? "ADDRESS" : "${data.terraform_remote_state.vpc.outputs.env}-ADDRESS")]
        }
      }
      behavior {
        name = "origin"
        option {
          key   = "cacheKeyHostname"
          value = "ORIGIN_HOSTNAME"
        }
        option {
          key   = "compress"
          value = true
        }
        option {
          key    = "customValidCnValues"
          values = ["{{Origin Hostname}}", "{{Forward Host Header}}"]
        }
        option {
          key   = "enableTrueClientIp"
          value = true
        }
        option {
          key   = "forwardHostHeader"
          value = "ORIGIN_HOSTNAME"
        }
        option {
          key   = "hostname"
          value = "${aws_cloudfront_distribution.s3_distribution.domain_name}"
        }
        option {
          key   = "httpPort"
          value = 80
        }
        option {
          key   = "httpsPort"
          value = 443
        }
        option {
          key   = "originCertificate"
          value = ""
        }
        option {
          key   = "originCertsToHonor"
          value = "STANDARD_CERTIFICATE_AUTHORITIES"
        }
        option {
          key   = "originSni"
          value = false
        }
        option {
          key   = "originType"
          value = "CUSTOMER"
        }
        option {
          key   = "ports"
          value = ""
        }
        option {
          key    = "standardCertificateAuthorities"
          values = ["akamai-permissive", "THIRD_PARTY_AMAZON"]
        }
        option {
          key   = "trueClientIpClientSetting"
          value = false
        }
        option {
          key   = "trueClientIpHeader"
          value = "True-Client-IP"
        }
        option {
          key   = "verificationMode"
          value = "CUSTOM"
        }
      }
    }
    rule {
      name = "Content Compression"
      criteria {
        name = "contentType"
        option {
          key   = "matchCaseSensitive"
          value = false
        }
        option {
          key   = "matchOperator"
          value = "IS_ONE_OF"
        }
        option {
          key   = "matchWildcard"
          value = true
        }
        option {
          key    = "values"
          values = ["text/*", "application/javascript", "application/x-javascript", "application/x-javascript*", "application/json", "application/x-json", "application/*+json", "application/*+xml", "application/text", "application/vnd.microsoft.icon", "application/vnd-ms-fontobject", "application/x-font-ttf", "application/x-font-opentype", "application/x-font-truetype", "application/xmlfont/eot", "application/xml", "font/opentype", "font/otf", "font/eot", "image/svg+xml", "image/vnd.microsoft.icon"]
        }
      }
      behavior {
        name = "gzipResponse"
        option {
          key   = "behavior"
          value = "ALWAYS"
        }
      }
    }
    rule {
      name = "Static Content"
      criteria {
        name = "fileExtension"
        option {
          key   = "matchCaseSensitive"
          value = false
        }
        option {
          key   = "matchOperator"
          value = "IS_ONE_OF"
        }
        option {
          key    = "values"
          values = ["aif", "aiff", "au", "avi", "bin", "bmp", "cab", "carb", "cct", "cdf", "class", "css", "doc", "dcr", "dtd", "exe", "flv", "gcf", "gff", "gif", "grv", "hdml", "hqx", "ico", "ini", "jpeg", "jpg", "js", "mov", "mp3", "nc", "pct", "pdf", "png", "ppc", "pws", "swa", "swf", "txt", "vbs", "w32", "wav", "wbmp", "wml", "wmlc", "wmls", "wmlsc", "xsd", "zip", "pict", "tif", "tiff", "mid", "midi", "ttf", "eot", "woff", "woff2", "otf", "svg", "svgz", "webp", "jxr", "jar", "jp2"]
        }
      }
      behavior {
        name = "caching"
        option {
          key   = "behavior"
          value = "CACHE_CONTROL_AND_EXPIRES"
        }
        option {
          key   = "mustRevalidate"
          value = false
        }
        option {
          key   = "defaultTtl"
          value = "1h"
        }
        option {
          key   = "honorPrivateEnabled"
          value = false
        }
        option {
          key   = "honorMustrevalidateEnabled"
          value = false
        }
      }
      behavior {
        name = "prefetch"
        option {
          key   = "enabled"
          value = false
        }
      }
      behavior {
        name = "prefetchable"
        option {
          key   = "enabled"
          value = true
        }
      }
    }
    rule {
      name = "Dynamic Content"
      criteria {
        name = "cacheability"
        option {
          key   = "matchOperator"
          value = "IS_NOT"
        }
        option {
          key   = "value"
          value = "CACHEABLE"
        }
      }
      behavior {
        name = "downstreamCache"
        option {
          key   = "behavior"
          value = "TUNNEL_ORIGIN"
        }
      }
    }
    rule {
      name = "Security Headers"
      behavior {
        name = "modifyOutgoingResponseHeader"
        option {
          key   = "action"
          value = "MODIFY"
        }
        option {
          key   = "standardModifyHeaderName"
          value = "OTHER"
        }
        option {
          key   = "newHeaderValue"
          value = "max-age=31536000;preload"
        }
        option {
          key   = "avoidDuplicateHeaders"
          value = true
        }
        option {
          key   = "customHeaderName"
          value = "Strict-Transport-Security"
        }
      }
      behavior {
        name = "modifyOutgoingResponseHeader"
        option {
          key   = "action"
          value = "MODIFY"
        }
        option {
          key   = "standardModifyHeaderName"
          value = "OTHER"
        }
        option {
          key   = "newHeaderValue"
          value = "nosniff"
        }
        option {
          key   = "avoidDuplicateHeaders"
          value = true
        }
        option {
          key   = "customHeaderName"
          value = "X-Content-Type-Options"
        }
      }
      behavior {
        name = "modifyOutgoingResponseHeader"
        option {
          key   = "action"
          value = "MODIFY"
        }
        option {
          key   = "standardModifyHeaderName"
          value = "OTHER"
        }
        option {
          key   = "newHeaderValue"
          value = "no-referrer-when-downgrade"
        }
        option {
          key   = "avoidDuplicateHeaders"
          value = true
        }
        option {
          key   = "customHeaderName"
          value = "Referrer-Policy"
        }
      }
      behavior {
        name = "modifyOutgoingResponseHeader"
        option {
          key   = "action"
          value = "MODIFY"
        }
        option {
          key   = "standardModifyHeaderName"
          value = "OTHER"
        }
        option {
          key   = "newHeaderValue"
          value = "1; mode=block"
        }
        option {
          key   = "avoidDuplicateHeaders"
          value = true
        }
        option {
          key   = "customHeaderName"
          value = "X-XSS-Protection"
        }
      }
      behavior {
        name = "modifyOutgoingResponseHeader"
        option {
          key   = "action"
          value = "MODIFY"
        }
        option {
          key   = "standardModifyHeaderName"
          value = "OTHER"
        }
        option {
          key   = "newHeaderValue"
          value = "SAMEORIGIN"
        }
        option {
          key   = "avoidDuplicateHeaders"
          value = true
        }
        option {
          key   = "customHeaderName"
          value = "X-FRAME-OPTIONS"
        }
      }
    }
  }
}


```

### Debug Output
[https://gist.github.com/mayaaravotwm/3b035efab591b19d590ad502f5d526c4](url)

### Expected Behavior
Not change those parameters every run

### Actual Behavior
Parameters change every run - it takes 15 minutes every build

### Steps to Reproduce
1. `terraform plan`

**Comment by martinstibbe** [comment]
@mayaaravotwm Thanks for reopening ticket I was able to locate source of difference coming up in rules will push through a fix in a release
Also ...

behavior {
      name = "tieredDistribution"
      option {
        key   = "enabled"
        value = true
      }
    }

 I get an error applying this rule

Rule validation error: https://problems.luna.akamaiapis.net/papi/v0/validation/attribute_required  The Tiered Distribution Map option on the `Tiered Distribution` behavior is required.

**Comment by mayaaravotwm** [comment]
Hey thanks for your help!
I actually don't know because it works for us.
When will the release be?

**Comment by mayaaravotwm** [comment]
hey, is there a update?

**Comment by lkowalsk-akamai-com** [comment]
Hi @mayaaravotwm , the fix is still being evaluated and waiting for inclusion. This takes that long as we are in the middle of bigger restructure of the provider. If all goes smoothly, the fix should be included in next release. I do not want to commit to any dates, but it should be ready within coming weeks.

**Comment by zstlaw** [comment]
Wanted to give an update on this ticket.  

There will be a 0.10 release this week but this issue was not contained in the 0.10 release.  This fix should be part of the 1.0.0 release where much of papi has been reworked to eliminate issues like this one.  The 1.0.0 release makes a lot of changes and it takes time for us to document them and the migration process properly.  We are testing to ensure this issue is resolved and hope to give you another update when we are sure it is completely ready.

**Comment by webuniverseio** [comment]
Hi, sorry for question which is not directly related to this issue 🙇 

I came here from google searching for "application/xmlfont/eot". I see that this configuration was added to my property configuration by Akamai support, and I see this content type mentioned here, also under compression. 

I wonder if it is a typo which is copied by support teams from some kind of template and this is a combination of "application/xml, font/eot" which is already listed in compression rules? "application/xmlfont/eot" is not a real type, right?

Again sorry and thank you 🙏

**Comment by zstlaw** [comment]
Thanks for reporting this issue. It should have been fixed in the 1.0.0 release. Please try upgrading to 1.0.0 and if you still face issues, Please let us know. Please refer https://registry.terraform.io/providers/akamai/akamai/latest/docs/guides/1.0_migration for additional details.

I can not personally answer the type question above but agree it looks like a concatenation error.
